wallet-new/test/Cardano/Wallet/WalletLayer/QuickCheck.hs

```diff
@@ -145,14 +145,11 @@ instance Arbitrary GetUtxosError where
                       , GetUtxosErrorFromGetAccountsError <$> arbitrary
                       ]
 
-genTxIn :: Gen TxIn
-genTxIn = oneof
-    [ TxInUtxo <$> arbitrary <*> arbitrary
-    , TxInUnknown <$> choose (1, 255) <*> scale (min 150) arbitrary
-    ]
-
 instance Arbitrary TxIn where
-    arbitrary = genTxIn
+    arbitrary = oneof
+        [ TxInUtxo <$> arbitrary <*> arbitrary
+        , TxInUnknown <$> choose (1, 255) <*> scale (min 150) arbitrary
```

Comment:
I'm not sure this is what you want to do. This will set the `size` parameter to QuickCheck generation to be the minimum of what it already is or 150. If you want to restrict the actual values, then you'd want `suchThat`:

```haskell
arbitrary `suchThat` (<= 150)
```

Would generate valid values up to 150.

Reply:
Here I trusted what Core people used in core/test/Test/Pos/Core/Arbitrary/Txp.hs . Which is of course not a proof that it is correct. I will contact Eric concerning this. Maybe it will give rise to a new ticket and I inform him to affect also this instance